MonoGame.Framework/Audio/SoundEffect.XAudio.cs

```diff
@@ -139,41 +139,42 @@ private void PlatformInitialize(byte[] buffer, int sampleRate, AudioChannels cha
         {
             CreateBuffers(  new WaveFormat(sampleRate, (int)channels),
                             DataStream.Create(buffer, true, false),
-                            0, 
-                            buffer.Length);
+                            0);
```

Comment:
For sure this code is bugged...  it is passing loop length as the buffer length which is incorrect.  This code should be correctly calculating the length in samples and passing it down.